### PR TITLE
Support composite uniqueness constraints from Cypher

### PR DESCRIPTION
**Pull request by craigtaverner**
Includes support for constraint population and verification

changelog: Support composite uniqueness constraints from Cypher, including creation, dropping and verification

### CODE REVIEW

**Comment by systay** [comment]
Cypher code LGTM! 👍

File: community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/schema/verification/CompositeDuplicateCheckingCollector.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                else if ( propertyValuesEqual( properties, currentValues ) )
+                {
+                    throw new IndexEntryConflictException( current.nodeId[i], nodeId,
+                            OrderedPropertyValues.ofUndefined( currentValues ) );
+                }
+            }
+            current = current.next;
+        }
+        while ( current != null );
+    }
```
Comment (chrisvest) [inline community/lucene-index/src/main/java/org/neo4j/kernel/api/impl/schema/verification/CompositeDuplicateCheckingCollector.java]: I wonder if there's more duplication between this class and its parent, that could be removed here.

**Comment by chrisvest** [comment]
Kernel code looks good to me. The test failures don't appear to be related, so I queued up another build.

**Comment by chrisvest** [comment]
@craigtaverner Please rebase. Also, the build failed on a `ReadReplicaHierarchicalCatchupIT`, which sounds like something that could be flaky, or (hopefully) fixed already.

**Comment by craigtaverner** [comment]
This work is a subset of https://github.com/neo4j/neo4j/pull/9046. As such we need only merge https://github.com/neo4j/neo4j/pull/9046 in order to get both in. Alternatively, merge this one, then rebase the other and merge it later. To save time, I'd recommend merging only https://github.com/neo4j/neo4j/pull/9046.

**Comment by craigtaverner** [comment]
We've decided to stop rebasing both #9046 and #9002 and just close this PR and merge the other.

**Pull request closed without merging**